compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -4728,27 +4728,30 @@ public BType getXMLSequenceType(BType xmlSubType) {
 
     public void visit(BLangXMLSequenceLiteral bLangXMLSequenceLiteral) {
         BType expType = types.getReferredType(this.expType);
-        if (expType.tag != TypeTags.XML && expType.tag != TypeTags.UNION && expType.tag != TypeTags.XML_TEXT
-        && expType != symTable.noType) {
-            dlog.error(bLangXMLSequenceLiteral.pos, DiagnosticErrorCode.INCOMPATIBLE_TYPES, this.expType,
-                    "XML Sequence");
-            resultType = symTable.semanticError;
-            return;
-        }
-
-        List<BType> xmlTypesInSequence = new ArrayList<>();
+        Set<BType> xmlTypesInSequence = new LinkedHashSet<>();
 
         for (BLangExpression expressionItem : bLangXMLSequenceLiteral.xmlItems) {
             resultType = checkExpr(expressionItem, env, this.expType);
-            if (!xmlTypesInSequence.contains(resultType)) {
+            if (resultType != symTable.semanticError) {
                 xmlTypesInSequence.add(resultType);
             }
         }
-
+        if (xmlTypesInSequence.isEmpty()) {
+            resultType = symTable.semanticError;
+            return;
+        }
+        // Disallow exp type that has 'xml:T (singleton) items
+        if (expType.tag == TypeTags.XML_ELEMENT || expType.tag == TypeTags.XML_COMMENT
+                || expType.tag == TypeTags.XML_PI) {
+            dlog.error(bLangXMLSequenceLiteral.pos, DiagnosticErrorCode.INCOMPATIBLE_TYPES, this.expType,
+                    "XML Sequence");
```

Comment:
Why are we hard-coding  "XML Sequence"? We should use whatever the static type. Saying  "XML Sequence" doesn't quite help the user IMO.

Can we please fix this also?